src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -334,6 +340,8 @@ private boolean isDesiredError(AsyncResult<Response> reply, String errMsg) {
       }
     } else if (reply.cause() instanceof PgException) {
       return PgExceptionUtil.get(reply.cause(), 'D').matches(errMsg);
+    } else if (reply.cause().getMessage() != null) {
```

Comment:
Exception can be created without cause, propose to rewrite to:
Objects.notNull(reply.cause()) && StringUtils.isNotEmpty(reply.cause().getMessage())

Reply:
Refactored